client/src/pages/DuelCreate.jsx

```diff
@@ -56,38 +82,66 @@ const DuelCreate = ({ history }) => {
           type="text"
           placeholder="Rechercher un utilisateur"
           onChange={(event) => {
+            event.target.value === "" ? setSearch(null) : setSearch(true);
             setSelected(null);
-            setSearchRegex(new RegExp(event.target.value, "i"));
+            setSearchRegex(new RegExp(event.target.value, "y"));
           }}
         ></input>
       </section>
 
       <section>
-        <ul>
-          {Object.keys(listOfUsers)
-            .filter((pseudo) => !searchRegex || searchRegex.test(pseudo))
-            .map((pseudo) => {
-              const user = listOfUsers[pseudo];
-              return (
-                <li
-                  key={user.pseudo}
-                  onClick={() => setSelected(user.pseudo)}
-                  onDoubleClick={() => createDuel(user.pseudo)}
-                  className={selected === user.pseudo ? "selected" : ""}
-                >
-                  <Avatar size="50px" infos={user.avatar} />
-                  <p>{user.pseudo}</p>
-                </li>
-              );
-            })}
-        </ul>
+        {search ? (
+          <ul>
+            {Object.keys(listOfUsers)
+              .filter((pseudo) => !searchRegex || searchRegex.test(pseudo))
+              .map((pseudo) => {
+                const user = listOfUsers[pseudo];
+                return (
+                  <li
+                    key={user.pseudo}
+                    onClick={() => setSelected(user.pseudo)}
+                    onDoubleClick={() => createDuel(user.pseudo)}
+                    className={selected === user.pseudo ? "selected" : ""}
+                  >
+                    <Avatar size="50px" infos={user.avatar} />
+                    <p>{user.pseudo}</p>
+                  </li>
+                );
+              })}
+          </ul>
+        ) : (
+          <></>
+        )}
+        <div id="filtres">
+          <h2> Niveau des questions : </h2>
+          <label>
+            <input onChange={changeDifficulty} type="radio" name="difficulty" value={0} />
```

Comment:
Ici on veut que l'input soit coché si la difficulté vaut 0
```suggestion
            <input onChange={changeDifficulty} type="radio" name="difficulty" value={0} checked={difficulty === 0}  />
```